.github/workflows/publish-helm.yml

```diff
@@ -0,0 +1,25 @@
+name: "Publish: helm chart"
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'helm/**'
+      - '.github/workflows/publish-helm.yml'
+jobs:
+  publish-release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Helm
+        uses: azure/setup-helm@v1
+        with:
+          version: v3.5.2
+
+      - name: Publish Helm charts
+        uses: stefanprodan/helm-gh-pages@master
```

Comment:
I would recommend using the official helm publisher action: https://github.com/helm/chart-releaser-action